### PR TITLE
Flake8 E712 fix

### PR DESCRIPTION
**Pull request by khabdrick**
Four flake8 tests clean up E712

### CODE REVIEW

File: infogami/infobase/_dbstore/store.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def _get_doc(self, row):
         return doc
 
     def put(self, key, doc):
-        if doc.get("_delete") == True:
+        if doc.get("_delete"):
             return self.delete(key, doc.get("_rev"))
```
Comment (anandology) [inline infogami/infobase/_dbstore/store.py:76]: That looks okay, but both have different behavior. The initial code accepts only `True` as a value for `_delete` where as the new one accepts any non-null string or non-zero integer. Do we really want to make these changes in the name of Flake8, without worrying about consequences?

File: infogami/utils/markdown/markdown.py [hunk 2]
```diff
@@ -248,10 +248,11 @@ def unlink(self) :
...
 
     def setAttribute(self, attr, value) :
-        if not attr in self.attributes :
-            self.attributes.append(attr)
+        if attr in self.attributes :
+            self.attribute_values[attr] = value
 
-        self.attribute_values[attr] = value
+        
+        self.attributes.append(attr)
```
Comment (anandology) [inline infogami/utils/markdown/markdown.py:255]: The behavior is completely changed. We'll now hav duplicate vlaues in the `attributes` when setAttribute is called multiple times with the same attr.

**Review by anandology — COMMENTED** [review]
I'm not a big fan of making code changes to a codebase with a long history with a suggestion of an automated tool. I'm scared that this may lead to quite unexpected bugs. Treat it with care.

File: infogami/utils/markdown/markdown.py (hunk 2 above)
Comment (khabdrick) [reply infogami/utils/markdown/markdown.py]: I have made the appropriate changes

File: infogami/infobase/_dbstore/store.py (hunk 1 above)
Comment (cclauss) [reply infogami/infobase/_dbstore/store.py:76]: Please revert this change.  This PR should only be modifying lines that flake8 is complaining about.

File: infogami/infobase/dbstore.py [hunk 3]
```diff
@@ -440,7 +440,7 @@ def get_id(key):
                     # 'bot' column is not enabled
                     if key == 'bot' and not config.use_bot_column:
                         bots = get_bot_users(self.db)
-                        if value or str(value).lower() == "true":
+                        if value is True or str(value).lower() == "true":
```
Comment (cclauss) [inline infogami/infobase/dbstore.py:443]: Please revert this change.

File: infogami/utils/template.py [hunk 4]
```diff
@@ -76,7 +76,7 @@ def __getitem__(self, name):
         t = dict.__getitem__(self, name)
         if isinstance(t, LazyTemplate):
             t = t.func()
-        elif web.config.debug and self.is_template_modified(t):
+        elif web.config.debug is True and self.is_template_modified(t):
```
Comment (cclauss) [inline infogami/utils/template.py:79]: Please revert this change.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: infogami/infobase/dbstore.py (hunk 3 above)
Comment (khabdrick) [reply infogami/infobase/dbstore.py:443]: flake8 actually complained about those lines that's why I touched it

Comment (cclauss) [reply infogami/infobase/dbstore.py:443]: This change is OK now.

File: infogami/utils/markdown/markdown.py [hunk 5]
```diff
@@ -248,9 +248,11 @@ def unlink(self) :
         self.childNodes = None
 
     def setAttribute(self, attr, value) :
-        if not attr in self.attributes :
+        if attr not in self.attributes :
             self.attributes.append(attr)
+
```
Comment (cclauss) [inline infogami/utils/markdown/markdown.py:253]: Please revert the adding of this blank line.

File: infogami/utils/markdown/markdown.py [hunk 6]
```diff
@@ -248,9 +248,11 @@ def unlink(self) :
         self.childNodes = None
 
     def setAttribute(self, attr, value) :
-        if not attr in self.attributes :
+        if attr not in self.attributes :
             self.attributes.append(attr)
+            
 
+
```
Comment (cclauss) [inline infogami/utils/markdown/markdown.py:255]: Please revert the adding of this blank line.

File: infogami/utils/template.py (hunk 4 above)
Comment (cclauss) [reply infogami/utils/template.py:79]: This is good now.

File: infogami/utils/markdown/markdown.py [hunk 7]
```diff
@@ -1570,7 +1572,7 @@ def _applyPattern(self, line, pattern) :
 
         if isinstance(node, Element):
 
-            if not node.nodeName in ["code", "pre"] :
+            if  node.nodeName not in ["code", "pre"] :
```
Comment (cclauss) [inline infogami/utils/markdown/markdown.py:1575]:
```suggestion
            if node.nodeName not in ("code", "pre"):
```

File: infogami/utils/markdown/markdown.py [hunk 8]
```diff
@@ -248,9 +248,9 @@ def unlink(self) :
         self.childNodes = None
 
     def setAttribute(self, attr, value) :
-        if not attr in self.attributes :
-            self.attributes.append(attr)
-
+        if attr not in self.attributes :
+            self.attributes.append(attr)       
+
```
Comment (cclauss) [inline infogami/utils/markdown/markdown.py:253]:
```suggestion
            self.attributes.append(attr)

```

File: infogami/utils/markdown/markdown.py [hunk 9]
```diff
@@ -248,7 +248,7 @@ def unlink(self) :
         self.childNodes = None
 
     def setAttribute(self, attr, value) :
-        if not attr in self.attributes :
+        if attr not in self.attributes :
```
Comment (cclauss) [inline infogami/utils/markdown/markdown.py:251]:
```suggestion
        if attr not in self.attributes:
```

**Review by cclauss — APPROVED** [review]
